PrebidFledge.md

```diff
@@ -0,0 +1,185 @@
+
+### Winning Bid
+
+This could be as simple as a numeric value or it could be as rich as an OpenRTB bid response object. I think it depends on how complex the Prebid IG Controller is and what info it needs.
+
+```javascript
+var prebidWinningConextualBid = {
+    // possibly an OpenRTB Bid Response object, though doesn't have to be
+}
+```
+
+### Auction Configuration
+
+Fledge offers a flexible configuration object which can be used to pass arbitrary information to the buyers bidding function (perBuyerSignals), the sellers scoring function (sellerSignals), or both (auctionSignals).
+
+Ideally this would be represented as an OpenRTB bid request that is supplemented with SSP specific metadata where needed. An OpenRTB bid request is a fairly standard way of capturing contextual information that many SSPs are familiar with and is reasonably concise. It's also consistent with how many of the SSPs probably received the contextual ad request.
+
+```javascript
+const myAuctionConfig = {
+  'seller': 'publisher.com',
+  'decisionLogicUrl': '/publishersPrebidIGBundle.js', // link to publisher's prebidIG bundle
+  'trustedScoringSignalsUrl': 'publishers-prebid-server.com/trusted-scoring-signals/',
+  'interestGroupBuyers': ['www.example-dsp.com', 'buyer2.com', ...],
+  'additionalBids': [ prebidWinningContextualBid ],
+  'auctionSignals': ortbBidRequest,   // captures contextual information
+  'sellerSignals': {
+    sspMetadata: {  // metadata provided by contextual adapters
+      "magnite": {...},
+      "openx": {...},
+      "pubmatic": {...}
+    }
+  },
+  'perBuyerSignals': {
+    'www.example-dsp.com': {...},
+    'www.another-buyer.com': {...},
+    ...},
+};
+```
+
+The invocation would then look like:
+
+```javascript
+navigator.runAdAuction(myAuctionConfig).then((auctionResult) => {
+  if (auctionResult) {
+    // IG ad rendered
+  } else {
+    // Render the contextual ad
+  }
+})
+```
+
+## Open Questions
+
+1. Is there a permissions model for access to IG Bids? What is it?
+An SSP specific TD API seems impossible (see below). The browser running the auction seems bad (see Parrrot). Having bids available for rendering by anyone seems problematic as well. This proposal assumes that anyone who invokes renderInterestGroupAd has access to all bids. I suppose SSPs could encrypt their responses and then pass a key into the metadata object that would allow only them to decrypt it. This actually seems like it might be the best way, though passing an encryption key through the wild is only mildly secure so it's not real security.
+
+2. How will GAM interact with the TD API?
+How will GAM integrate into the final selection layer in the browser and will it help or hinder SSPs? If GAM were to provide an an API/adapter that integrated with Prebid.js this would be a good thing. If, on the other hand, GAM provides a tag library that directly invokes ```navigator.renderInterestGroupAd``` without allowing other SSPs to participate this would be bad.
```

Comment:
```suggestion
How will GAM integrate into the final selection layer in the browser and will it help or hinder SSPs? If GAM were to provide an API/adapter that integrated with Prebid.js this would be a good thing. If, on the other hand, GAM provides a tag library that directly invokes `navigator.renderInterestGroupAd` without allowing other SSPs to participate this would be bad.
```